csm/core/controllers/information.py

```diff
@@ -77,3 +77,115 @@ async def post(self):
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        Log.info(f"Fetching cortx topology for resource:{resource} and resource_id:{resource}.")
+        # Call Cortx Information Service
+        response = await self._service.get_specific_resource(resource, resource_id)
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}/{view}")
+class AllViews(CsmView):
+    """
+    Get information about specific resource from cortx topology.
+    GET: Get information about specific resource from cortx topology.
+    """
+    def __init__(self, request):
+        super().__init__(request)
+        self._service =self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query information about all
+           views of specific resource from cortx topology."""
+        Log.debug("Handling GET request to query information about all views of specific resource from cortx topology.")
+        # Read path parameter
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        view = self.request.match_info[const.ARG_VIEW]
+        # Call Cortx Information Service
+        Log.info(f"Fetching cortx topology for resource:{resource}, resource_id:{resource} and view:{view}.")
+        response = await self._service.get_views(resource, resource_id, view)
+        return CsmResponse(response)
+
+@CsmAuth.public
+@CsmView._app_routes.view("/api/v2/system/topology/{resource}/{resource_id}/{view}/{view_id}")
+class SpecificView(CsmView):
+    """GET REST implementation to query information about specific
+       views of specific resource from cortx topology."""
+    Log.debug("Handling GET request to fetch details from .")
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app[const.INFORMATION_SERVICE]
+
+    async def get(self):
+        """GET REST implementation to query only specific subset of CORTX topology."""
+        Log.debug("Handling GET request to query only specific subset of CORTX topology.")
+        # Read path parameter
+        resource = self.request.match_info[const.ARG_RESOURCE]
+        resource_id = self.request.match_info[const.ARG_RESOURCE_ID]
+        view = self.request.match_info[const.ARG_VIEW]
+        view_id = self.request.match_info[const.ARG_VIEW_ID]
+        path_params_dict = {
+            const.ARG_RESOURCE : resource,
+            const.ARG_RESOURCE_ID : resource_id,
+            const.ARG_VIEW : view,
+            const.ARG_VIEW_ID : view_id
+        }
+        Log.info(f"Fetching cortx topology for {path_params_dict}.")
+        # Call Cortx Information Service
+        response = await self._service.get_specific_view(**path_params_dict)
```

Comment:
No resource/view validation error is handled here. Please add those

Reply:
Added resource/view validation error handling